sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/DataSourceStrategy.scala

```diff
@@ -500,11 +569,16 @@ private[sql] object DataSourceStrategy extends Strategy with Logging {
     // For conciseness, all Catalyst filter expressions of type `expressions.Expression` below are
     // called `predicate`s, while all data source filters of type `sources.Filter` are simply called
     // `filter`s.
+    var jdbcRelation = false
```

Comment:
You can do it like:

    val jdbcRelation = relation match {
      case jdbc: JDBCRelation => jdbcRelation = true
      case _ => jdbcRelation = false
    }